e2e/tests/wasm/grandpa_test.go

```diff
@@ -0,0 +1,407 @@
+package wasm
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/icza/dyno"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/polkadot"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/relayer"
+	"github.com/strangelove-ventures/interchaintest/v8/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	testifysuite "github.com/stretchr/testify/suite"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	transfertypes "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+)
+
+func TestGrandpaTestSuite(t *testing.T) {
+	testifysuite.Run(t, new(GrandpaTestSuite))
+}
+
+type GrandpaTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+const (
+	heightDelta      = uint64(20)
+	votingPeriod     = "30s"
+	maxDepositPeriod = "10s"
+)
+
+// TestGrandpaContract features
+// * sets up a Polkadot parachain
+// * sets up a Cosmos chain
+// * sets up the Hyperspace relayer
+// * Funds a user wallet on both chains
+// * Pushes a wasm client contract to the Cosmos chain
+// * create client, connection, and channel in relayer
+// * start relayer
+// * send transfer over ibc
+func (s *GrandpaTestSuite) TestGrandpaContract() {
```

Comment:
Maybe a more specific test name, since it would be good to have also another one that exercises a timeout.

```suggestion
func (s *GrandpaTestSuite) TestMsgTransfer_Succeeds_GrandpaContract() {
```

Reply:
are you happy if I do this change in the follow up?